autoscale/src/main/java/com/sequenceiq/periscope/monitor/CleanupMonitor.java

```diff
@@ -0,0 +1,44 @@
+package com.sequenceiq.periscope.monitor;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.periscope.model.ScalingActivities;
+import com.sequenceiq.periscope.monitor.evaluator.EvaluatorExecutor;
+import com.sequenceiq.periscope.monitor.evaluator.cleanup.ActivityCleanupEvaluator;
+
+@Component
+@ConditionalOnProperty(prefix = "periscope.enabledAutoscaleMonitors.cleanup-monitor", name = "enabled", havingValue = "true")
+public class CleanupMonitor extends ScalingActivityMonitor {
+
+    private static final long CLEAN_UP_DURATION_HOURS = 24L;
+
+    @Override
+    protected List<ScalingActivities> getMonitored() {
+        Set<Long> activityIdsToCleanup = getScalingActivityService().findAllScalingActivityIdsThatEndedBefore(CLEAN_UP_DURATION_HOURS);
```

Comment:
end_time or start_time?
Will there be a way where we can filter by status? Will need to be careful about entities which are potentially in a stuck state.

Reply:
Updated to start_time & added a status filter here as well considering these two fields are indexed in the DB for faster retrieval